ta/remoteproc/include/elf_parser.h

```diff
@@ -24,6 +24,12 @@
  *
  * This structure shall be consistent with the Linux kernel structure
  * definition from include/linux/remoteproc.h.
+ *
+ * The resource_table structure is stored in a shared memory, accessed by the
```

Comment:
If this is the only reason for the packed attribute then you could just add a `static_assert(sizeof(struct resource_table) == 4 * sizeof(uint32_t))` instead. As a nice bonus, you'd allow the compile to generate efficient code.

However, I suspect that in the kernel the struct may end up at an unaligned address or offset and that's the main reason for the `__packed` attribute. If this can't happen in OP-TEE then there's no need for a `__packed` attribute that only causes confusion and inefficient code. If it can happen, just mention how and I'm sure the `__packed` attribute will make sense to a reader.

Code is much easier to understand when you don't add things for good measure or because it was added somewhere else for reasons unknown.

Reply:
I reanalyze the use cases. The Linux Kernel  parses the structure to get resources declared just after the `resource_table` structure in memory. It gets the resource addresses thanks to the `offset` field. The size of the `offset` field depends on the number of resources declared in `num` field.  In this case the __packed makes sense to avoid memory corruption, as the memory region is structured in several structures not 64-bit aligned.

The OP-TEE doesn't parse the resources. It just gets the start address of the structure and performs some basic checks on the `resource_table` structure. The size is retrieved from the `.resource_table` elf section.

So just removing the `__packed ` attribute seems OK. Adding the test with  `static_assert` seems useless and not accurate  due to the variability of offset field size.
I will test it and reworks the fixup

Thanks for your advise!

Reply:
I reworked the last fixup! ( force update) to remove the "__packed" and add a related comment

Reply:
OK, thanks.